test/e2e/storage/pv_protection.go

```diff
@@ -83,9 +84,16 @@ var _ = utils.SIGDescribe("PV Protection", func() {
 		framework.ExpectNoError(framework.WaitForPersistentVolumePhase(v1.VolumeAvailable, client, pv.Name, 1*time.Second, 30*time.Second))
 
 		By("Checking that PV Protection finalizer is set")
-		pv, err = client.CoreV1().PersistentVolumes().Get(pv.Name, metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred(), "While getting PV status")
-		Expect(slice.ContainsString(pv.ObjectMeta.Finalizers, volumeutil.PVProtectionFinalizer, nil)).To(BeTrue(), "PV Protection finalizer(%v) is not set in %v", volumeutil.PVProtectionFinalizer, pv.ObjectMeta.Finalizers)
+		err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
```

Comment:
We should move this to its own function, so as e2e test flow is more natural to read.

Reply:
Thanks for your review, that makes sense.